pkg/apierror/const.go

```diff
@@ -0,0 +1,63 @@
+//------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------
+
+package apierror
+
+// ErrorCategory indicates the kind of error
+type ErrorCategory string
+
+const (
+	// ClientError is expected error
+	ClientError ErrorCategory = "ClientError"
+
+	// InternalError is system or internal error
+	InternalError ErrorCategory = "InternalError"
+)
+
+// Common Azure Resource Provider API error code
+type ErrorCode string
+
+const (
+	// From Microsoft.Azure.ResourceProvider.API.ErrorCode
+	InvalidParameter                      ErrorCode = "InvalidParameter"
+	BadRequest                            ErrorCode = "BadRequest"
+	NotFound                              ErrorCode = "NotFound"
+	Conflict                              ErrorCode = "Conflict"
+	PreconditionFailed                    ErrorCode = "PreconditionFailed"
+	OperationNotAllowed                   ErrorCode = "OperationNotAllowed"
+	OperationPreempted                    ErrorCode = "OperationPreempted"
+	PropertyChangeNotAllowed              ErrorCode = "PropertyChangeNotAllowed"
+	InternalOperationError                ErrorCode = "InternalOperationError"
+	InvalidSubscriptionStateTransition    ErrorCode = "InvalidSubscriptionStateTransition"
+	UnregisterWithResourcesNotAllowed     ErrorCode = "UnregisterWithResourcesNotAllowed"
+	InvalidParameterConflictingProperties ErrorCode = "InvalidParameterConflictingProperties"
+	SubscriptionNotRegistered             ErrorCode = "SubscriptionNotRegistered"
+	ConflictingUserInput                  ErrorCode = "ConflictingUserInput"
+	ProvisioningInternalError             ErrorCode = "ProvisioningInternalError"
+	ProvisioningFailed                    ErrorCode = "ProvisioningFailed"
+	NetworkingInternalOperationError      ErrorCode = "NetworkingInternalOperationError"
+	QuotaExceeded                         ErrorCode = "QuotaExceeded"
+	Unauthorized                          ErrorCode = "Unauthorized"
+	ResourcesOverConstrained              ErrorCode = "ResourcesOverConstrained"
+	ControlPlaneProvisioningInternalError ErrorCode = "ControlPlaneProvisioningInternalError"
+	ControlPlaneProvisioningSyncError     ErrorCode = "ControlPlaneProvisioningSyncError"
+	ControlPlaneInternalError             ErrorCode = "ControlPlaneInternalError"
+	ControlPlaneCloudProviderNotSet       ErrorCode = "ControlPlaneCloudProviderNotSet"
+
+	// From Microsoft.WindowsAzure.ContainerService.API.AcsErrorCode
```

Comment:
the errors from here down appear to be RP errors. I'm not sure any of them apply in the ACS-Engine scenario. We should have a const file in the RP with these values and make sure that the datamodel down here just stores a string and matches these that way we can set values not defined here.